client/src/components/Visualization/charts/Geochart.js

```diff
@@ -1,59 +1,67 @@
 /**
  * Description: Geochart implements a Google geochart component.
  * Author: Kira Toal
- * Date: 2020/07/13
+ * Date: 2020/07/21
  */
-
-import React, { useEffect, useState } from 'react';
-import firebase, { app, database } from '../../../firebase/firebase';
-
-import { Chart } from 'react-google-charts';
-import { states } from './StateDataParser';
+import { Chart } from "react-google-charts";
+import React, { useState, useEffect } from 'react';
+import { STATES } from '../../../constants/geochart_constants';
+import { database } from '../../../firebase/firebase';
 
 const Geochart = () => {
+  const STATES_COLLECTION = 'dev_states';
   const [adTotal, setAdTotal] = useState(0);
   const WIDTH = '700';
   const HEIGHT = '400';
 
-  class GeochartAd {
-    constructor(id) {
-      /**
-       * @param {string} id
-       */
-      this.id = id;
-    }
-    toString() {
-      return this.id;
-    }
-  }
+  useEffect(() => {
+    async function fetchStateTotals() {
+      /* 
+      dataTable is read by the Chart component into Geochart format.
+      The states are colored according to the value in the second column of
+        the table ("Total Ad Spend (USD)").
+      The first column (state name) and third column (info on total state ad 
+        spend and top advertiser) are shown in the tooltip.
+      */ 
+      let dataTable = [["State", "Total Ad Spend (USD)", {type: 'string', role: 'tooltip'}]];
 
-  // Firestore data converter converts snapshots to custom objects.
-  const adConverter = {
-    toFirestore(ad) {
-      return {
-        id: ad.id,
-      };
-    },
-    fromFirestore(snapshot, options) {
-      const data = snapshot.data(options);
-      return new GeochartAd(data.id);
-    },
-  };
+      let stateSpendPromises = [];
+      let topAdvertiserPromises = [];
+
+      for (let state of STATES) {
+        stateSpendPromises.push(database.collection(STATES_COLLECTION)
+            .doc(state.toLowerCase())
+            .get()
+        );
+        topAdvertiserPromises.push(database.collection(STATES_COLLECTION)
+            .doc(state.toLowerCase())
+            .collection('advertisers')
+            .orderBy('totalAdvertiserSpend')
+            .limit(1)
+            .get()
+        );
+      }
+
+      let stateResults = await Promise.all(stateSpendPromises);
+      let advertiserResults = await Promise.all(topAdvertiserPromises);
+      const zippedResults = stateResults.map((e, i) => [e,advertiserResults[i]]);
 
-  useEffect(async () => {
-    const data = [['State', 'Total Ad Spend Across All Advertisers']];
-    for (const state in states) {
-      data.push([
-        state,
-        Math.floor(Math.random() * Math.floor(250000) + 10000),
-      ]); // Update data table.
+      for (let result of zippedResults) {
```

Comment:
You can make use of ES6 destructuring here.

`for (let [stateTotal, topAdvertiser] of zippedResults) {`